### ISSUE TITLE
Other models available other than 4-stacked?

### ISSUE THREAD

**Issue by Yozey** [1adrianb/face-alignment#4]
Hello Adrian,

Thanks for sharing your code in pytorch. I noticed that you have several models with less stacked model here:
[https://github.com/1adrianb/face-alignment/blob/478c38823785db76ca85c059cd455a23a586b663/face_alignment/api.py#L26]

However, only 4-stacked pretrained model is available to download from your website. I would like to know that if it is possible to share smaller models even though the performance will be a little bit worse.

Thank you very much in advance.

**Comment by 1adrianb** [comment]
Hi @Yozey ,

Yes, I was in the process of converting/retraining the smaller models but most of them are not fully retrained. I will resume this at some point during the next week and add them here. Sorry for the delay.

**Comment by ggsonic** [comment]
Hi @1adrianb , will the 2dfan-2.pth.tar be available? i want to have a test. or will you please send me a 2-stacked 2dfan model to my gmail ggsonic@gmail.com. Thanks!

**Comment by korabelnikov** [comment]
Any changes in subject? Does not-stacked model available now?

**Comment by 1adrianb** [comment]
Hi @nightmaredimple ,
The entire process is dominated by dlib face detector. In CPU mode dlib uses basically a V-J face detector while in cuda mode a cnn-based detector.The 2D/2.5D detector alone (ignoring the face detection) should be able to run at 20-30fps.  A simple way to speedup the process is to use a faster face detector. Another method of speeding this up is using a batch>1.

**Comment by nightmaredimple** [comment]
@1adrianb  Thanks for your apply！I‘ve done a simple test(**detect_landmarks_in_image.py**),the results as below:
In GPU mode:
face detection:0.46s
2D alignment:3.22s
2D-to-3D alignment:2.04s
Total time:5.8s
then the same detected face for the second alignment:
2D alignment:0.08s
2D-to-3D alignment:0.1s
so make the alignment twice can speed up!

In CPU mode:
face detection:0.15s
2D alignment:2.22s
2D-to-3D alignment:1.10s
Total time:3.4s
then the same detected face for the second alignment:
2D alignment:2.34s
2D-to-3D alignment:1.355s
so the GPU can speed up the alignment process

emmm, maybe my GPU (Tesla K40c) can't make it real-time，but I still appreciate for your help.By the way,is there any more models can be shared except for 2DFAN-4 and 3DFAN-4?

**Comment by Wuyboo** [comment]
Just cannot download 2D-FAN model from  https://www.adrianbulat.com/face-alignment/ 
emmm...thank you anyway.

**Comment by xinguo2015** [comment]
Hi @1adrianb, I am trying the 2D alignment, however, but it seems have problem to download the model: https://www.adrianbulat.com/downloads/python-fan/2DFAN-4.pth.tar
Is the model not available anymore?

**Comment by 1adrianb** [comment]
Hi @nightmaredimple, that makes sense, the first pass is taking longer due to all initialisation and copying stuff from cpu to cuda. 
The 20-30 was achieved usign a Titan X Pascal, however a simple way to speed up is to use a 2 stack network (basically 2x improvment with almost no loss in performance), however this will require retraining the network for best performance.  Another suggestion is to try to export the network to a static graph (via onnx) wich should hopefully fuse some operations and improve speed. I will probaly push this change at a later point.

Not at this unfortuantelly.

**Comment by njho** [comment]
Hi @1adrianb I'd also be interested in seeing the 2 stack network. Any update? That being said, great work!

**Comment by moncio** [comment]
Hi @1adrianb , any kind of advance in this task? thank you so much

**Comment by moncio** [comment]
Sorry for being heavy @1adrianb  again, but I am very interested to know about this topic if there is any progress. Thanks in advance.

**Comment by BigTom007** [comment]
Hi @1adrianb , I'm also very interested about 2 stack network, since 2 stack Hourglass was used in your other paper.
